.github/workflows/docker_build.yml

```diff
@@ -0,0 +1,67 @@
+---
+name: Run precommit and conditionally build container
+
+on:
+  push:
+    branches:
+      - '*'
+    tags:
+      - 'v[0-9]+\.[0-9]+\.[0-9]+'
+  pull_request:
+    branches:
+      - '*'
+
+concurrency: ci-${{ github.ref }}
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_PATH: ghcr.io/${{ github.repository }}
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Static Analysis
+        uses: pre-commit/action@v3.0.0
```

Comment:
i learned recently that you really only need to specify the major version numbers on all of these actions, i.e. `pre-commit/action@v3`, which will make it easier to maintain.

Reply:
Doesn't look like it:

> Error: Unable to resolve action `pre-commit/action@v3`, unable to find version `v3`

Reply:
hmm, i guess not everyone follows [GH best practices on release tags](https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-tags-for-release-management).